### ISSUE TITLE
New block options

### ISSUE THREAD

**Issue by stefan-cotitosu** [Codeinwp/otter-blocks#20]
### Button Group  - it would be great if we can have the following options

- [x]   - line-height control for the buttons text
- [x]   - responsive option, to have the buttons one above each other on mobile
- [ ]   - add responsive alignment

### Advanced heading

- Responsiveness settings toggle works only for the font-size, it would be great to work for the line-height too, or for all typography settings. (Hardeep's note: Not important, will see if more people request as discussed with Stefan)

- [x]   - font-size is limited to 200px
- [x]   - `font-weight: regular;` it gives me an error in browser, it says invalid property; from what I read it doesn't exist, but there is `font-weight: normal;`
- rgba control for color for text color (Hardeep's note: Not important, will see if more people request as discussed with Stefan, or if Gutenberg enables the disableAlpha prop in future)
- [x]  - add alignment control that would work separately on each device; this situation is very often, where text is aligned left on desktop & tablet and centered on mobile

### Section

- [x]   - for inner Columns, margin-left and margin-right control, sometimes I need negative margin (this is specific to Neve front page, with the negative margin)

### For Future
- [x] - Image Card Block
- [x]   - option to show/hide inner column, based on device; similar with Elementor, the idea is to be able to hide a column on tablet or mobile
- [x]  - padding/margin that will affect all columns; right now if I add 4 columns, by default they all have margin and padding 20px instead of 0px; I have to edit each column to clear that(including mobile devices); I would like a control that can affect all of them, and if the inner column set the properties, the column setting will have higher priority; this generic control should be placed in Section layout near the `Columns gap` drop-down.

**Comment by ineagu** [comment]
@stefan-cotitosu but what about putting the block group inside a nested 1 column container block, which has margin control and so on?

**Comment by stefan-cotitosu** [comment]
@ineagu I will try a Section block with 2 columns and a Button group block with only one button in each column. 

This should work, thanks 👍

**Comment by abaicus** [comment]
A suggestion that I want to add here: 

- [ ] It'd be useful if `Maximum Content Width` for the Section Structure could have responsive options.

**Comment by HardeepAsrani** [comment]
```padding/margin that will affect all columns; right now if I add 4 columns, by default they all have margin and padding 20px instead of 0px; I have to edit each column to clear that(including mobile devices); I would like a control that can affect all of them, and if the inner column set the properties, the column setting will have higher priority; this generic control should be placed in Section layout near the Columns gap drop-down.```

@abaicus @radu-themeisle @rodica-andronache @stefan-cotitosu I guess we can have an option to link all values together? Like to link the change of Mobile, Tablet and Desktop together, to make things easier. What do you think about that?

**Comment by HardeepAsrani** [comment]
Closing the ticket and moving the unsorted items to their own tickets.

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Can we add `line-height` control based on the device?

The font-size control can have a different value on each device. That's great but I want to be able to change the line-height too.

Here is a situation with 
 - `font-size: 60px;` on Desktop
 - `font-size: 37px;` on Mobile

with the same line-height, I have to choose
 - very large space between rows on mobile
 - text collapsed on Desktop

In order to solve this I left the `line-height` value empty, so it will have a value in `em` unit.

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani here is a feedback after the first test on the Neve front page sections.
I am writing this so you don't have to gather info from the pull requests comments, hope it helps :)

- [ ] - add line-height control based on device. If I don't set a line-height, everything looks good until changing the theme. This is a particular case because Neve adds line-height in em.
- [ ] - fix the word-break issue, if it's possible, in the next release, today or this week, not sure when you schedule it.
- [ ] - for the social icons group, the icons order is different in the editor vs front-end. https://prnt.sc/mdrz0d https://prnt.sc/mdrzcx
- [ ] - please check this
I also have this problem, when adding two testimonials sections for example, and try to edit one, the other one is also updated, and the block controls get duplicated. If you don't have this problem, maybe somebody else can try to replicate it. I saw it on my local and poppy too.
https://screencast-o-matic.com/watch/cqVUIZ3yT0
- [ ] - adding an image block with border, padding and background overlay options. I tried to achieve a transparent border for an image and I failed because of the image caption, which adds a bottom margin, so I have to change many things, like padding, background color, background overlay on other elements, to achieve something simple

**Comment by HardeepAsrani** [comment]
> adding an image block with border, padding and background overlay options. I tried to achieve a transparent border for an image and I failed because of the image caption, which adds a bottom margin, so I have to change many things, like padding, background color, background overlay on other elements, to achieve something simple

You mean to add a new block, right?

**Comment by HardeepAsrani** [comment]
> I also have this problem, when adding two testimonials sections for example, and try to edit one, the other one is also updated, and the block controls get duplicated. If you don't have this problem, maybe somebody else can try to replicate it. I saw it on my local and poppy too.

Nice catch! Seems like this happens with the unique ID coming from clientId. This is major.

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Can we remove the drag & drop padding control from the Section block and inner columns.

I find it confusing, because the structure is complex with just 2 inner columns, and the default values for padding and margin are still 20px. I set them to 0, then I accidentally moved that padding line and it seems that nothing was changed. Later I discovered that the paddings are not 0px anymore.
Maybe it's just me.

What do you think @rodica-andronache @radu-themeisle @abaicus ?

**Comment by radu-themeisle** [comment]
It's useful, no need to remove it but rather fix it as it doesn't update in real-time and it's confusing. Here's a screen recording and what's interesting is that one of the default blocks update the padding value as you drag: https://www.useloom.com/share/c5006d0989f14ce08da43c2a699cf6ec

**Comment by selul** [comment]
@stefan-cotitosu @radu-themeisle while the feedback is welcomed and would be addressed in the future, you need to develop the templates with the current otter's features, find solutions that do not involve releasing a new version each time you find a bug or a suggestion. 

These things you have reported, they will be analyzed and gathered in releases but you don't need to count on them when doing your work. 

You need to find workarounds, bundle fixes in theme or something similar.

Long story short: Please do report each time you find something wrong or something that could have been improved with Otter, but manage to finish your task without counting on that thing.

**Comment by stefan-cotitosu** [comment]
@selul Thanks for the advice. We are always finish the tasks in a way or another with the current options of Otter. 

I just try to note down suggestion and ask for feedback here, to have the data centralized. Otherwise, it will be hard for Hardeep to gather them from other issues and pull requests discussions.

And sometimes I need a second opinion, not sure if my approach is the best and should be implemented 🙂

**Comment by radu-themeisle** [comment]
- add responsive alignment for buttons: it would be good to be able to align buttons to left/right on desktop/tablet and center them on mobile

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Do you think we can add background color for Advanced Heading?

I am trying to build this card a little bit differently. Left card is the new version.
![new-pricing-card](https://user-images.githubusercontent.com/26867648/52119548-0ff57980-2622-11e9-826b-4e691e96ec45.png)

The current solution is to wrap the two headings in a section so I can add the background color on the section or inner column element.
In such situations, by creating many containers such as Section-Column block, or Service block to wrap things so it can easily be copied, the difference between front-end and the editor increases. Each container (I mean the blocks mentioned above) gives you the possibility of adding other blocks inside it. Because of that, the empty paragraph block at the end of each container makes an unused space higher. 

Here is the editor/front-end difference that I want to highlight.
<img width="1053" alt="headings-with-wrapper" src="https://user-images.githubusercontent.com/26867648/52119935-7fb83400-2623-11e9-9900-daae7c99e830.png">

I consider this an improvement for the plugin, despite the fact that I am now using it for Neve front page. What do you think?

**Comment by HardeepAsrani** [comment]
@stefan-cotitosu 

1. It's not possible because heading will not have a background like that, it will have to be wrapped in a div which won't make sense, so best way is to wrap is around a column and hit backspace on the last paragraph. (I know it's not intuitive but that's Gutenberg :) )

2. This is indeed a bug which doesn't have an easy solution so far, given how weird the things are nested inside Editor. 😞  Hopefully, in a future update I can learn Flexbox more deeply and implement a better CSS.

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Thanks for the update. Let me know if I can help you with centering the section content using flexbox or another approach. Maybe together we can deal with it. 🙂

**Comment by HardeepAsrani** [comment]
I don't think another approach will be better, Flexbox seems to be the best way to do it. We can revisit this after 1.2 and fix this together. 😄

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Can we add some options to the Post Grid block?

I've created a very simple blog section for Neve front page. 
![image](https://user-images.githubusercontent.com/26867648/52342985-8bd73380-2a1f-11e9-98d7-0f2e889ff344.png)

Didn't use the post thumbnail because I find the image too big. I know that a control for the image size might lead to an image not aligned with the content (excerpt). Still think that we need an image size control.

Also, it would be great to be able to center the text on mobile and tablet, the whole post content, title, meta, excerpt. 
![image](https://user-images.githubusercontent.com/26867648/52343126-e5d7f900-2a1f-11e9-925c-6d49e5e2b041.png)

Post title color is inherited from the theme. In Neve it looks blue because Neve makes all links blue by default. Can we add a color control for the titles in the Post Grid block? 🙂 

Let me know your thoughts 👍

**Comment by HardeepAsrani** [comment]
@stefan-cotitosu Yes, we have planned on redoing it in 1.3, along side Google Maps block.

**Comment by stefan-cotitosu** [comment]
@HardeepAsrani Thanks. I guess this section will look better when this is done. For now I've finished it in a simple way 🙂

**Comment by HardeepAsrani** [comment]
Closing this as it has become too long. All the relevant issues are noted in other places. If you've more issues/feature requests, make new issues so it's easy to track.
